### ISSUE TITLE
New rule: `svelte/no-literal-id`

### ISSUE THREAD

**Issue by Turbo87** [sveltejs/eslint-plugin-svelte#1577]
### Motivation

Literal `id` attributes in reusable components can produce duplicate IDs when multiple instances are rendered. Svelte’s scoped CSS does not scope the IDs themselves.

For example, a component containing `<label for="email">` and `<input id="email">` works in isolation. Rendering it twice causes both labels to reference the first input, so clicking the second label focuses the wrong field. Duplicate IDs can also break relationships expressed through `aria-labelledby` and `aria-describedby`.

Svelte 5.20 introduced [`$props.id()`](https://svelte.dev/docs/svelte/$props#props.id()), which provides an ID unique to each component instance. Svelte guarantees that its value is consistent between server rendering and client hydration. Using it to construct IDs avoids collisions between component instances.

### Description

The rule should catch plain attributes such as `id="email"` and equivalent string expressions such as `id={'email'}` and recommend deriving component-local IDs from `$props.id()`. Dynamic IDs, including IDs supplied through props, should be allowed without requiring the rule to trace their origin.

### Examples

```svelte
<script>
  const uid = $props.id();
  let { id } = $props();
</script>

<!-- ✓ GOOD -->
<label for="{uid}-email">Email</label>
<input id="{uid}-email" type="email" />

<p id="{uid}-description">Additional information</p>

<div {id}></div>

<!-- ✗ BAD -->
<input id="email" type="email" />
<input id={'email'} type="email" />
<input id={`email`} type="email" />
```

### Additional comments

An automatic fix would need caution because references to the ID may exist outside the component.

The documentation should clarify that `$props.id()` is unique per component instance. Multiple elements within that instance still need distinct suffixes, including items rendered in an `{#each}` block.

Related:
- #1106
- sveltejs/svelte#6932

**Comment by Turbo87** [comment]
FWIW I'm happy to attempt an implementation, if this rule is deemed useful :)
